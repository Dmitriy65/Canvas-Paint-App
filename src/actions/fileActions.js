import FileUtility from '../utils/FileUtility';
export const HANDLE_FILE_SUCCESS = 'HANDLE_FILE_SUCCESS';
export const HANDLE_FILE_ERROR = 'HANDLE_FILE_ERROR';

export const handleInputFile = (file) => {
  return async dispatch => {

    const [outputLink, canvas, error] = await FileUtility.processInputFile(file);

    if (error) dispatch(errorFileProcess(error));
    else dispatch(successFileProcess([outputLink, canvas]));
  }
}

export const successFileProcess = (fileProps) => {
  return {
    type: HANDLE_FILE_SUCCESS,
    payload: fileProps
  }
}

export const errorFileProcess = (errorMessage) => {
  return {
    type: HANDLE_FILE_ERROR,
    payload: errorMessage
  }
}