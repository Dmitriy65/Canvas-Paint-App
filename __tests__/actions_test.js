import configureMockStore from 'redux-mock-store';
import "babel-polyfill"
import { successFileProcess, errorFileProcess, HANDLE_FILE_SUCCESS, HANDLE_FILE_ERROR } from '../src/actions/fileActions';
import { stringCanvas } from './CanvasField_test';
const mockStore = configureMockStore();
const store = mockStore({});

describe('action creators', () => {
  it('creates successFileProcess action when file reading was success', () => {

    store.dispatch(successFileProcess(
      {
        type: HANDLE_FILE_SUCCESS,
        payload: [
          'blob:http://localhost:3000/ddafffc8-1d42-4cee-a186-1077686179f4',
          stringCanvas
        ]
      }
    ));
    expect(store.getActions()).toMatchSnapshot();
  });

  it('creates errorFileProcess action when file reading was not success', () => {

    store.dispatch(errorFileProcess(
      {
        type: HANDLE_FILE_ERROR,
        payload: 'Cant read input file, type doesn`t match "text/plain"'
      }
    ));
    expect(store.getActions()).toMatchSnapshot();
  });
});
