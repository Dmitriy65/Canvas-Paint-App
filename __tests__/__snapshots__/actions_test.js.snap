// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`action creators creates errorFileProcess action when file reading was not success 1`] = `
Array [
  Object {
    "payload": Object {
      "payload": Array [
        "blob:http://localhost:3000/ddafffc8-1d42-4cee-a186-1077686179f4",
        "----------------------
|oooooooooooooooxxxxx|
|xxxxxxooooooooox   x|
|     xoooooooooxxxxx|
|     xoooooooooooooo|
----------------------",
      ],
      "type": "HANDLE_FILE_SUCCESS",
    },
    "type": "HANDLE_FILE_SUCCESS",
  },
  Object {
    "payload": Object {
      "payload": "Cant read input file, type doesn\`t match \\"text/plain\\"",
      "type": "HANDLE_FILE_ERROR",
    },
    "type": "HANDLE_FILE_ERROR",
  },
]
`;

exports[`action creators creates successFileProcess action when file reading was success 1`] = `
Array [
  Object {
    "payload": Object {
      "payload": Array [
        "blob:http://localhost:3000/ddafffc8-1d42-4cee-a186-1077686179f4",
        "----------------------
|oooooooooooooooxxxxx|
|xxxxxxooooooooox   x|
|     xoooooooooxxxxx|
|     xoooooooooooooo|
----------------------",
      ],
      "type": "HANDLE_FILE_SUCCESS",
    },
    "type": "HANDLE_FILE_SUCCESS",
  },
]
`;
